.circleci/config.yml

```diff
@@ -12,12 +15,24 @@ workflows:
     jobs:
       - build:
           context:
-            - npm
+            - npm-readonly
+            - github
+            - dockerhub
+      - publish:
```

Comment:
I broke `publish` into a separate step that will run after `build` completes on `main`. This allows branches to use the read-only context, and the publish step to use the publisher context.